### PR TITLE
Bug 1218537 - Replace Job Info artifacts with a "jobdetail" endpoint

### PR DESCRIPTION
**Pull request by wlach**
This is simpler and removes one more type of artifact from the per-project
databases

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1410)

### CODE REVIEW

**Comment by jgraham** [comment]
Review status: 0 of 20 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[treeherder/jobs/__init__.py, line 0 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/1410#-KFOt6HRldPVk77breTB:-KFOt6HSeAGlyBcBCc4m:1016843252) ([raw file](https://github.com/mozilla/treeherder/blob/4a357b297fde5d5ba3f93c27a53aea53292f53a9/treeherder/jobs/__init__.py#L0)):*
I feel like creating a whole new top level module just for the jobs stuff is overkill and doesn't really match the rest of the project. Is there any specific reason you chose this design?

---

*[treeherder/jobs/models.py, line 10 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/1410#-KFOvic6qxgwQD73ON3J:-KFOvic6qxgwQD73ON3K:-225998735) ([raw file](https://github.com/mozilla/treeherder/blob/dea948bb8e3d99ae83efde6b0b17e3b2151e0eae/treeherder/jobs/models.py#L10)):*
Does this map onto a char or a varchar in the database? I assume the latter, but if it's the former then the storage requirements seem unreasonable (3 bytes per possible character iiuc).

---

*[treeherder/jobs/models.py, line 11 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/1410#-KFOthEq-O62BcYD7Ugg:-KFOthEq-O62BcYD7Ugh:-1214425091) ([raw file](https://github.com/mozilla/treeherder/blob/dea948bb8e3d99ae83efde6b0b17e3b2151e0eae/treeherder/jobs/models.py#L11)):*
Do we have any evidence that 200 characters is enough length here?

---

*[treeherder/jobs/models.py, line 12 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/1410#-KFOtq-bYY_VQb_vzfWd:-KFOtq-c2lsnq9AMyPsW:-1977037556) ([raw file](https://github.com/mozilla/treeherder/blob/dea948bb8e3d99ae83efde6b0b17e3b2151e0eae/treeherder/jobs/models.py#L12)):*
Should there also be a content_type field or is it an intentional decision to get rid of that?

---

*[treeherder/webapp/api/jobs.py, line 266 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/1410#-KFOuzRTi_u2YpNKgxPz:-KFOuzRUDuQuJU-LfCly:106940769) ([raw file](https://github.com/mozilla/treeherder/blob/dea948bb8e3d99ae83efde6b0b17e3b2151e0eae/treeherder/webapp/api/jobs.py#L266)):*
I guess it doesn't matter for this case, because the number of entries is small, but in general we should try to use IdPagination.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/treeherder/1410)*

File: treeherder/jobs/models.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from django.db import models
+
+
+class JobDetail(models.Model):
+    '''
+    Represents metadata associated with a job
+    '''
+    id = models.AutoField(primary_key=True)
+    job_guid = models.CharField(max_length=50, db_index=True)
+    title = models.CharField(max_length=200, null=True)
```
Comment (camd) [inline treeherder/jobs/models.py]: In the UI html, it looks like it expects to always have a ``title`` and ``value`` but that ``url`` can be null.  Should the title not have the ``null=True`` here?

Comment (wlach) [inline treeherder/jobs/models.py]: In some cases we don't have a title, for example the "purged clobber" metadata here:

https://treeherder.mozilla.org/#/jobs?repo=mozilla-inbound&selectedJob=26197592

I will update the UI to reflect this.

**Comment by wlach** [comment]
Going to land this as part of a different bug, opened #1462

**Pull request closed without merging**